lib/floe/workflow/choice_rule/data.rb

```diff
@@ -81,11 +81,11 @@ def is_timestamp?(value) # rubocop:disable Naming/PredicateName
         end
 
         def compare_key
-          @compare_key ||= payload.keys.detect { |key| key.match?(/^(IsNull|IsPresent|IsNumeric|IsString|IsBoolean|IsTimestamp|String|Numeric|Boolean|Timestamp)/) }
+          @compare_key ||= (payload.keys - %w(Variable Next)).first
```

Comment:
I'm a little worried about moving from a list of known keys to everything except these two.  For one thing I think there is a Default key that is optional that we don't implement yet, and if someone mistype'd something it could get picked up here.

Would be nice to move this list of known keys to a constant and verify the payload on initialize instead of on evaluation like we have now.

Reply:
I'll pull out of this PR

Reply:
if someone forgets or mistypes the comparison operator, then our code blows up on a nil error

Reply:
https://github.com/ManageIQ/floe/pull/187